verda/clusters/_clusters.py

```diff
@@ -0,0 +1,285 @@
+        status: Current operational status of the cluster.
+        created_at: Timestamp of cluster creation.
+        location: Datacenter location code (default: Locations.FIN_03).
+        cluster_type: Type of the cluster.
+        worker_nodes: List of nodes in the cluster.
+        ssh_key_ids: List of SSH key IDs associated with the cluster nodes.
+        image: Image ID or type used for cluster nodes.
+        startup_script_id: ID of the startup script to run on nodes.
+        public_ip: IP address of the jumphost.
+    """
+
+    id: str
+    hostname: str
+    description: str
+    status: str
+    created_at: str
+    location: str
+    cluster_type: str
+    worker_nodes: list[ClusterWorkerNode]
+    ssh_key_ids: list[str]
+    image: str | None = None
+    startup_script_id: str | None = None
+    ip: str | None = None
+
+
+class ClustersService:
+    """Service for managing compute clusters through the API.
+
+    This service provides methods to create, retrieve, and manage compute clusters.
+    """
+
+    def __init__(self, http_client) -> None:
+        """Initializes the ClustersService with an HTTP client.
+
+        Args:
+            http_client: HTTP client for making API requests.
+        """
+        self._http_client = http_client
+
+    def get(self, status: str | None = None) -> list[Cluster]:
+        """Retrieves all clusters or clusters with specific status.
+
+        Args:
+            status: Optional status filter for clusters. If None, returns all
+                non-deleted clusters.
+
+        Returns:
+            List of cluster objects matching the criteria.
+        """
+        clusters_dict = self._http_client.get(CLUSTERS_ENDPOINT, params={'status': status}).json()
+        return [
+            Cluster.from_dict(cluster_dict, infer_missing=True) for cluster_dict in clusters_dict
+        ]
+
+    def get_by_id(self, id: str) -> Cluster:
+        """Retrieves a specific cluster by its ID.
+
+        Args:
+            id: Unique identifier of the cluster to retrieve.
+
+        Returns:
+            Cluster object with the specified ID.
+
+        Raises:
+            HTTPError: If the cluster is not found or other API error occurs.
+        """
+        cluster_dict = self._http_client.get(CLUSTERS_ENDPOINT + f'/{id}').json()
+        return Cluster.from_dict(cluster_dict, infer_missing=True)
+
+    def create(
+        self,
+        hostname: str,
+        cluster_type: str,
+        image: str,
+        description: str,
+        ssh_key_ids: list = [],
+        location: str = Locations.FIN_03,
+        startup_script_id: str | None = None,
+        shared_volume_name: str | None = None,
+        shared_volume_size: int | None = None,
+        *,
+        wait_for_status: str | None = ClusterStatus.PROVISIONING,
+        max_wait_time: float = 900,
+        initial_interval: float = 1.0,
+        max_interval: float = 10,
+        backoff_coefficient: float = 2.0,
```

Comment:
It's a new method, no need to keep backwards compatibility. Thus most arguments can be made keyword-only.

Let's also follow the same order - type, image, hostname - as in `.create` method for instances.

```suggestion
    def create(
        self,
        cluster_type: str,
        image: str,
        hostname: str,
        *,
        description: str,
        ssh_key_ids: list = [],
        location: str = Locations.FIN_03,
        startup_script_id: str | None = None,
        shared_volume_name: str | None = None,
        shared_volume_size: int | None = None,
        wait_for_status: str | None = ClusterStatus.PROVISIONING,
        max_wait_time: float = 900,
        initial_interval: float = 1.0,
        max_interval: float = 10,
        backoff_coefficient: float = 2.0,
```